docs/preview/features/secret-store/named-secret-providers.md

```diff
@@ -0,0 +1,68 @@
+---
+title: "Named secret providers"
+layout: default
+---
+
+# Retrieve a specific secret provider from the secret store
+
+The default workings of the secret store, is that a set of secret providers are registered and the consumer gets access to all of secrets provided by `ISecretProvider`s.
+In some cases, you may want to retrieve a specific secret provider from the store. This can be because that secret provider has some functionality that the other providers doesn't have.
+
+In those cases, you can register your secret provider with a unique name. This name will be matched with the secret provider so that in a later stage, you can retrieve back your named provider.
+
+## Register a named secret provider
+
+First, the secret provider has to be registered with a unique name. 
+Let's use this secret provider as an example:
+
+```csharp
+public class GetAndSetSecretprovider : ISecretProvider
```

Comment:
I think in this case adding a custom provider does not bring a lot of added value and might confuse people, no? Maybe just show how this works with the existing providers?

Reply:
Hmm, I thought with specifying a custom provider, it was more clear as of why you should use something that is already available in the 'default' secret store approach. 😅

Reply:
Well I think they usecase is regardless of custom/built-in. You define all the secret stores that you use but in some cases you need a specific one because you have to interact with that store for example.

Example - I need a device authentication key which has a dedicated vault, so you request that specific secret provider

Reply:
Aha, makes sense.

Reply:
Used the built-in environment variable secret provider. Is that better?

Reply:
Jup, thanks 👍